grpc/server.go

```diff
@@ -0,0 +1,87 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/agglayer/aggkit/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+type ServerConfig struct {
+	// Host is the address to bind the gRPC server
+	Host string `mapstructure:"Host"`
+	// Port is the port to bind the gRPC server
+	Port int `mapstructure:"Port"`
+	// EnableReflection indicates whether gRPC server reflection is enabled
+	EnableReflection bool `mapstructure:"EnableReflection"`
+}
+
+// Server encapsulates a gRPC server instance, its network listener, and the address it listens on.
+// It provides the necessary fields to manage the lifecycle and configuration of a gRPC server.
+type Server struct {
+	grpcServer *grpc.Server
+	listener   net.Listener
+	addr       string
+
+	cfg ServerConfig
+}
+
+// NewServer creates and returns a new gRPC Server instance listening on the specified address.
+// It accepts an address string, a boolean to enable server reflection, and optional gRPC server options.
+// If enableReflection is true, the server will register the reflection service for introspection.
+// Returns a pointer to the Server and an error if the listener cannot be created.
+func NewServer(cfg ServerConfig, opts ...grpc.ServerOption) (*Server, error) {
+	addr := fmt.Sprintf("%s:%d", cfg.Host, cfg.Port)
```

Comment:
Hmm I remember previously we were stripping the `http://` and `https://` prefixes https://github.com/agglayer/aggkit/blob/d34eb05ebb901737bcab1573b335262cb3089d5c/grpc/client.go#L254-L255

Is it relevant here as well?

Reply:
Fix: https://github.com/agglayer/aggkit/pull/711/commits/8426eb824e2e249d622e420bc3f19ae221fbf9a0